### PR TITLE
Update canonical build from source docs to reflect actual usage.

### PR DESCRIPTION
**Pull request by stellaraccident**
Both myself and our users have been falling down the hole of using an unsupported generator and fat fingering development options when we just want to be copy/pasting. I think how I have this laid out strike a balance.

### CODE REVIEW

File: docs/website/docs/building-from-source/getting-started.md [hunk 1]
```diff
@@ -44,8 +50,28 @@ git submodule update --init
...
+# Recommended for simple development using clang and lld:
+cmake -GNinja -B ../iree-build/ -H . \
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+    -DIREE_ENABLE_ASSERTIONS=ON \
+    -DCMAKE_C_COMPILER=clang \
+    -DCMAKE_CXX_COMPILER=clang++ \
+    -DCMAKE_LINKER=lld
+
+# Minimal/release with system compiler:
+cmake -GNinja -B ../iree-build/ -H . -DCMAKE_BUILD_TYPE=Release
```
Comment (benvanik) [inline docs/website/docs/building-from-source/getting-started.md:65]: you shouldn't need to specify either build type or ninja here (and without ninja multibuild configured having a different type here is an error) as those are active only during configuration above

File: docs/website/docs/building-from-source/getting-started.md [hunk 2]
```diff
@@ -11,19 +11,25 @@ compiler:
 
     <!-- TODO(scotttodd): annotation about gcc vs clang -->
 
-    ``` shell
-    sudo apt install cmake clang
```
Comment (benvanik) [inline docs/website/docs/building-from-source/getting-started.md:15]: I think it's worth having the copy paste commands here for cmake/ninja and if we are going to say clang below then we should have that too. I'd really like a CI for gcc though so we didn't have to specify.
```shell
python -m pip install ninja
sudo apt install cmake clang lld
```

Comment (stellaraccident) [reply docs/website/docs/building-from-source/getting-started.md:15]: Added back. Agreed on the CI. But I'd also like fewer users to show up in forums having gone down a rabbit hole on this.

File: docs/website/docs/building-from-source/getting-started.md (hunk 1 above)
Comment (stellaraccident) [reply docs/website/docs/building-from-source/getting-started.md]: Reworked.

**Review by benvanik — APPROVED** [review]

**Review by ScottTodd — APPROVED** [review]
Could also update the sample build commands in other files to include `-GNinja` for easier copy/paste.

I do wish the concise versions of these commands were good enough as defaults (e.g. just `bazel build iree/...` or `cmake --build`)... but the extra characters and complexity are probably worth it here :/

File: docs/website/docs/building-from-source/getting-started.md [hunk 3]
```diff
@@ -44,8 +55,28 @@ git submodule update --init
...
 
+Configure CMake:
+
+``` shell
+# Recommended for simple development using clang and lld:
+cmake -GNinja -B ../iree-build/ -S . \
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+    -DIREE_ENABLE_ASSERTIONS=ON \
+    -DCMAKE_C_COMPILER=clang \
+    -DCMAKE_CXX_COMPILER=clang++ \
```
Comment (ScottTodd) [inline docs/website/docs/building-from-source/getting-started.md:66]: (arg, GitHub ate my comment, retyping...)

We still want to recommend MSVC on Windows, right?

Could put this under a `=== "Linux and MacOS"` section or use the env var approach like before (suggest setting the env var above)
```
    export CC=clang
    export CXX=clang++
```

Comment (stellaraccident) [reply docs/website/docs/building-from-source/getting-started.md:66]: Good point. I can add a tab thingy for windows. I work on a lot of different projects and the env vars are killing me.

**Comment by stellaraccident** [comment]
> Could also update the sample build commands in other files to include `-GNinja` for easier copy/paste.
> 
> I do wish the concise versions of these commands were good enough as defaults (e.g. just `bazel build iree/...` or `cmake --build`)... but the extra characters and complexity are probably worth it here :/

I mean, they do/should work, but they funnel people into undefined territory that makes it hard to support them when they show up. I see it as a support cost trade-off.

**Comment by ScottTodd** [comment]
Pushed commits addressing my own comments.

**Pull request merged**